src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -54,31 +52,58 @@ class WorkflowActor private(binding: WdlBinding, actualInputs: Map[FullyQualifie
   // That which started the workflow, a role which has not yet been defined more specifically.
   private var _primeMover: Option[ActorRef] = None
 
-  private def isStarted: Boolean = _primeMover.isDefined
+  override def receive = notStarted
 
-  private def primeMover = _primeMover.get
+  private var maybeSymbolStore: Option[SymbolStore] = None
+  private var maybeExecutionStatusStore: Option[ExecutionStatusStore] = None
+  private var maybeBackend: Option[Backend] = None
+  private val log = Logging(context.system, this)
 
-  override def receive: Receive = {
+
+  private def notStarted: Receive = {
     LoggingReceive {
 
-      case Start =>
+      case Start(backend) =>
         _primeMover = Option(sender())
-        val executionStatusStore = new ExecutionStatusStore(binding)
-        val symbolStore = new SymbolStore(binding, actualInputs)
+        maybeExecutionStatusStore = Option(new ExecutionStatusStore(binding))
+        maybeSymbolStore = Option(new SymbolStore(binding, actualInputs))
+        maybeBackend = Option(backend)
         executionStatusStore.runnableCalls.foreach { call =>
-          startCallActor(symbolStore, call)
+          startCallActor(maybeSymbolStore.get, call)
         }
         primeMover ! Started
+        context.become(started)
 
+      case bad @ _ =>
+        sender ! InvalidOperation(s"Received message $bad before any Start message")
+    }
+  }
+
+  private def started: Receive = {
+    LoggingReceive {
       case CallActor.Started =>
       // TODO update execution status store
 
-      case CallActor.Done(taskOutputs) =>
-        // TODO Update execution status and symbol stores, evaluate workflow outputs
+      case CallActor.Done(call, callOutputs) =>
+        // TODO Update execution status and symbol stores.
         // TODO There is only one runnable task at the moment, so immediately
-        // TODO report as done to the creator.
-        primeMover ! Done(Map.empty[String, WdlValue])
-      // TODO This should reevaluate runnable tasks, if there are no more
+        // TODO report as done to the creator, passing back the symbol store.
+        val addedEntries = callOutputs.map { callOutput =>
+          // TODO very broken, assumes there are always output values here, but other code
+          // TODO has them properly Option'ed.
+          // TODO This should be messaging a symbol store actor to synchronize updates
+          symbolStore.addOutputValue(call.fullyQualifiedName, callOutput._1, Some(callOutput._2), callOutput._2.wdlType)
+        }
+
+        val failures = addedEntries.filter(_.isFailure)
```

Comment:
See my earlier comment. I think you should either use partition or use a collect to pull out the messages and map over the remainder if there are no msgs